### PR TITLE
Add  SonarQube to CI

### LINKED ISSUE #7881: Integrate airbyte repo with SonarQube
## Tell us about the problem you're trying to solve

Many issues we have nowadays can be avoided even with static analysis. 

We're going to evaluate the code quality using [SonarQube](https://www.sonarqube.org/). Using that tool we would be able to see all connectors' info in one place and have a good overview of the project code quality.

## Describe the solution you’d like

We're going to generate the SonarQube analytics, base on our tests, pylint and other inner code analytics tools.

To do that we need to implement the following steps:
- Fix github CI to make each job run proper and stable (for now it could fail with unknown reasons, so we could not rely on it).
- Paste together all possible code validation reports (such as pylint, unit tests, flake etc.) and generate the SonarQube assessments based on it. So we don't need to run the code evaluation from the SonarQube itself, just pass the reports there.

### Subtasks

- [x] Search the modules which were changed in PR
- [x] Integrate https://github.com/airbytehq/airbyte/pull/8873 static code checkers
- [x] Update static code checkers to allow building the reports appliable to SonarQube
- [x] Build reports for the modules found
- [x] Update PR after https://github.com/airbytehq/airbyte/pull/8873 is merged
- [x] Add unit tests for each module

## Describe the alternative you’ve considered or used
No alternative with given flow

### PR DESCRIPTION
**Pull request by h-chlor**
Using the SonarQube as tool for aggregation of all issues, bugs, vulnerabilities, coverage reports.  GitHub CI should run all tests and necessary scripts automatically and add summary reports as a new comment of source PR.
 
An UML draft of this solution is [here](https://whimsical.com/ci-sonarqube-v1-TMMHusnuau3Hm51CwuYmWn).

Now we are trying to cover Python and Java connectors only. But in the future this logic can be expanded for other languages and project components/modules.

 The following files were updated for testing of issues and changes detections. Please ignore these because they will be reverted before merging of this PR.
- airbyte-integrations/connectors/destination-mongodb/*
- airbyte-integrations/connectors/source-airtable/*
- airbyte-integrations/connectors/source-github/*

You can see examples of generated reports into comments of this PR. This report is generated by our script and we can edit and format its template anytime if needed

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/8362)

### CODE REVIEW

**Comment by antixar** [comment]
test

**Review by sherifnada — COMMENTED** [review]
@h-chlor could you add an `.md` file to the docs under `contributing to airbyte` section explaining at a high level how sonarQube is integrated? e.g: where is it hosted, how is it run/triggered, etc... basically if you were onboarding someone into how to use sonarqube, what info do they need to know to become effective at working with it?

**Comment by antixar** [comment]
> @h-chlor could you add an `.md` file to the docs under `contributing to airbyte` section explaining at a high level how sonarQube is integrated? e.g: where is it hosted, how is it run/triggered, etc... basically if you were onboarding someone into how to use sonarqube, what info do they need to know to become effective at working with it?

Hi @sherifnada , @h-chlor had initialised this PR only and I finished it.  DevOps team prepared [a similar doc](https://github.com/airbytehq/airbyte-cloud/wiki/IAP-tunnel-to-the-SonarQube-instance)
About docs did you validate my[ UML diagram ](https://whimsical.com/ci-sonarqube-v1-TMMHusnuau3Hm51CwuYmWn)? I didn't write some official documentation because I added several parts of this solution myself and I'm not sure to their are relevant and useful. I'm gonna add docs after approval of Airbyte team only.

**Comment by sherifnada** [comment]
@antixar indeed, the UML diagram is great! I think it's worth screenshotting + leaving a link to the whimsical in an `.md` doc in the repo itself, just to have a clear entrypoint for people. Also adding any links to relevant files/entrypoints etc..

I wasn't able to look at the code in details, but I think having something like this is a great insurance policy to help people make modifications to the code in the future

File: docs/contributing-to-airbyte/sonar-qube-workflow.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# SonarQube workflow
+
+## Goals
+&nbsp;Code of the Airbyte monorepo are contributed by a lot of developers. And human failures are no way around while merging of PRs.
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/sonar-qube-workflow.md:4]:
```suggestion
&nbsp;The Airbyte monorepo receives contributions from a lot of developers, and there is no way around human errors while merging PRs.
```

File: docs/contributing-to-airbyte/sonar-qube-workflow.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Likely every language has different tools for testing and validation of source files. The best practice is their using before pushing to git branches.
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/sonar-qube-workflow.md:5]:
```suggestion
Likely every language has different tools for testing and validation of source files. And while it's best practice to lint and validate code before pushing to git branches, it doesn't always happen.
```

File: docs/contributing-to-airbyte/sonar-qube-workflow.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# SonarQube workflow
+
+## Goals
+&nbsp;Code of the Airbyte monorepo are contributed by a lot of developers. And human failures are no way around while merging of PRs.
+Likely every language has different tools for testing and validation of source files. The best practice is their using before pushing to git branches. 
+But it is optional, and as rule as we detect possible problems after launch test/publish commands only. Therefore, using of automated CI code validation can  provided the following benefits:
+* Resultant problem/vulnerability reports would be useful when PR was created only. And developers would fix bugs and remove smells before code reviews.
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/sonar-qube-workflow.md:7]:
```suggestion
* Problem/vulnerability reports available when the PR was created. And developers would fix bugs and remove smells before code reviews.
```

File: docs/contributing-to-airbyte/sonar-qube-workflow.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+* Reviewers would be sure of all standard checking were made and code changes satisfy the requirements.
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/sonar-qube-workflow.md:8]:
```suggestion
* Reviewers would be sure all standard checks were made and code changes satisfy the requirements.
```

File: docs/contributing-to-airbyte/sonar-qube-workflow.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+### Python
+* [flake8](https://flake8.pycqa.org/en/stable/)
+* [mypy](https://mypy.readthedocs.io/en/stable/)
+* [isort](https://pycqa.github.io/isort/)
+* [black](https://black.readthedocs.io/en/stable/)
+* [coverage](https://coverage.readthedocs.io/en/6.2/)
+
+All Python tools use the common [pyproject.toml](https://github.com/airbytehq/airbyte/blob/master/pyproject.toml) file.
+
+### Common tools
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/sonar-qube-workflow.md:27]: Can you add a section describing to a developer where to find the SQ configuration? Let's say I'm someone who knows sonarqube exists in the repo but wants to follow the code lines very closely, where do I go?

**Review by sherifnada — APPROVED** [review]
Docs look great! Excited to have it in the repo. Just a few comments then feel free to merge.

File: docs/contributing-to-airbyte/sonar-qube-workflow.md (hunk 5 above)
Comment (antixar) [reply docs/contributing-to-airbyte/sonar-qube-workflow.md:27]: sure, added

**Comment by antixar** [comment]
Follow-up issues:
[Java coverage](https://github.com/airbytehq/airbyte/issues/9613)
[CDK + SAT detection](https://github.com/airbytehq/airbyte/issues/9612)

**Pull request merged**